presto-native-execution/presto_cpp/main/connectors/tpcds/TpcdsGen.cpp

```diff
@@ -0,0 +1,975 @@
+      return "customer_address";
+    case Table::TBL_CUSTOMER_DEMOGRAPHICS:
+      return "customer_demographics";
+    case Table::TBL_DATE_DIM:
+      return "date_dim";
+    case Table::TBL_HOUSEHOLD_DEMOGRAPHICS:
+      return "household_demographics";
+    case Table::TBL_INCOME_BAND:
+      return "income_band";
+    case Table::TBL_INVENTORY:
+      return "inventory";
+    case Table::TBL_ITEM:
+      return "item";
+    case Table::TBL_PROMOTION:
+      return "promotion";
+    case Table::TBL_REASON:
+      return "reason";
+    case Table::TBL_SHIP_MODE:
+      return "ship_mode";
+    case Table::TBL_STORE:
+      return "store";
+    case Table::TBL_STORE_RETURNS:
+      return "store_returns";
+    case Table::TBL_STORE_SALES:
+      return "store_sales";
+    case Table::TBL_TIME_DIM:
+      return "time_dim";
+    case Table::TBL_WAREHOUSE:
+      return "warehouse";
+    case Table::TBL_WEB_PAGE:
+      return "web_page";
+    case Table::TBL_WEB_RETURNS:
+      return "web_returns";
+    case Table::TBL_WEB_SALES:
+      return "web_sales";
+    case Table::TBL_WEB_SITE:
+      return "web_site";
+    default:
+      VELOX_UNREACHABLE();
+  }
+}
+
+TypePtr resolveTpcdsColumn(Table table, const std::string& columnName) {
+  return getTableSchema(table)->findChild(columnName);
+}
+
+Table fromTableName(const std::string_view& tableName) {
+  static std::unordered_map<std::string_view, Table> map{
+      {"call_center", Table::TBL_CALL_CENTER},
+      {"catalog_page", Table::TBL_CATALOG_PAGE},
+      {"catalog_returns", Table::TBL_CATALOG_RETURNS},
+      {"catalog_sales", Table::TBL_CATALOG_SALES},
+      {"customer", Table::TBL_CUSTOMER},
+      {"customer_address", Table::TBL_CUSTOMER_ADDRESS},
+      {"customer_demographics", Table::TBL_CUSTOMER_DEMOGRAPHICS},
+      {"date_dim", Table::TBL_DATE_DIM},
+      {"household_demographics", Table::TBL_HOUSEHOLD_DEMOGRAPHICS},
+      {"income_band", Table::TBL_INCOME_BAND},
+      {"inventory", Table::TBL_INVENTORY},
+      {"item", Table::TBL_ITEM},
+      {"promotion", Table::TBL_PROMOTION},
+      {"reason", Table::TBL_REASON},
+      {"ship_mode", Table::TBL_SHIP_MODE},
+      {"store", Table::TBL_STORE},
+      {"store_returns", Table::TBL_STORE_RETURNS},
+      {"store_sales", Table::TBL_STORE_SALES},
+      {"time_dim", Table::TBL_TIME_DIM},
+      {"warehouse", Table::TBL_WAREHOUSE},
+      {"web_page", Table::TBL_WEB_PAGE},
+      {"web_returns", Table::TBL_WEB_RETURNS},
+      {"web_sales", Table::TBL_WEB_SALES},
+      {"web_site", Table::TBL_WEB_SITE},
+  };
+
+  auto it = map.find(tableName);
+  if (it != map.end()) {
+    return it->second;
+  }
+  throw std::invalid_argument(
+      fmt::format("Invalid TPC-DS table name: '{}'", tableName));
+}
+
+RowVectorPtr genTpcdsData(
+    Table table,
+    size_t maxRows,
+    size_t offset,
+    memory::MemoryPool* pool,
+    double scaleFactor,
+    vector_size_t parallel,
+    vector_size_t child) {
+  switch (table) {
```

Comment:
It might be simpler to assign 2 variables for parentTable and childTable in this switch/case and have a single genTpcdsParentAndChildTable call at the end of the function.

Reply:
I assigned 2 variables for parentTable and childTable, but still have 3 calls to the `genTpcdsParentAndChildTable`. One workaround for having a single call could be:

  ```
    Table parentTable, childTable;
    bool isDependentTable;
    case Table::TBL_CATALOG_RETURNS:
    case Table::TBL_CATALOG_SALES:
      parentTable = Table::TBL_CATALOG_SALES;
      childTable = Table::TBL_CATALOG_RETURNS;
      isDependentTable= true;
      
    if(isDependentTable) {
      genTpcdsParentAndChildTable()// function call.
    }
      
```

Will this work?

Reply:
The current code is fine.